common/http_test.go

```diff
@@ -0,0 +1,245 @@
+package common_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tmlibs/common"
+)
+
+func TestWriteSuccess(t *testing.T) {
+	w := httptest.NewRecorder()
+	common.WriteSuccess(w, "foo")
+	assert.Equal(t, w.Code, 200, "should get a 200")
+}
+
+var blankErrResponse = new(common.ErrorResponse)
+
+func TestWriteError(t *testing.T) {
+	tests := [...]struct {
+		msg  string
+		code int
+	}{
+		0: {
+			msg:  "this is a message",
+			code: 419,
+		},
+	}
+
+	for i, tt := range tests {
+		w := httptest.NewRecorder()
+		msg := tt.msg
+
+		// First check without a defined code, should send back a 400
+		common.WriteError(w, errors.New(msg))
+		assert.Equal(t, w.Code, http.StatusBadRequest, fmt.Sprintf("#%d: should get a 400", i))
+		blob, err := ioutil.ReadAll(w.Body)
+		if err != nil {
+			assert.Failf(t, "expecting a successful ioutil.ReadAll", "#%d", i)
+			continue
+		}
+
+		recv := new(common.ErrorResponse)
+		if err := json.Unmarshal(blob, recv); err != nil {
+			assert.Failf(t, "expecting a successful json.Unmarshal", "#%d", i)
+			continue
+		}
+
+		assert.Equal(t, reflect.DeepEqual(recv, blankErrResponse), false, "expecting a non-blank error response")
+
+		// Now test with an error that's .HTTPCode() int conforming
+
+		// Reset w
+		w = httptest.NewRecorder()
+
+		common.WriteError(w, common.ErrorWithCode(errors.New("foo"), tt.code))
+		assert.Equal(t, w.Code, tt.code, fmt.Sprintf("#%d: got=%d want=%d", i, w.Code, tt.code))
```

Comment:
Good test, but you can write the check shorter.
1) assert.Equal always prints the two values if they don't equal, as a default.
2) The arguments after the comparison values are interpreter by fmt.Sprintf already

This is effectively the same:
`assert.Equal(t, w.Code, tt.code, "#%d", i)`

It will `fmt.Sprintf("#%d", i)` as the message for you, no need to write it out.
And want and got are always in the top of the error message.